enterprise/cli/provisionerdaemonstart.go

```diff
@@ -104,6 +104,23 @@ func (r *RootCmd) provisionerDaemonStart() *serpent.Command {
 				return err
 			}
 
+			displayedTags := make(map[string]string, len(tags))
+			for key, val := range tags {
+				displayedTags[key] = val
+			}
+
+			if provisionerKey != "" {
+				pkDetails, err := client.GetProvisionerKey(ctx, provisionerKey)
+				if err != nil {
+					return xerrors.New("unable to get provisioner key details")
+				}
+
+				displayedTags = make(map[string]string, len(pkDetails.Tags))
```

Comment:
What is your intention here? To clear out the map if the provisioner key is set?

Reply:
So globally, either we have no key and some tags (line 107 to 110) - either we have no key (line 112) and want to fill the displayedTags with the ones fetched in `pkDetails`.

That's why at first I was using the same map without allocating it , as we can't really be sure about the size before knowing in which case we'll enter 
anyway, we want the map to be declared in the whole function to be used below.

Reply:
Seems like you should add an `else` block which populates `displayedTags` if the provision key is nil?

I don't think what you have is a big deal performance-wise since maps can never shrink, so if the `make` call on L107 has a length of 4 and the call on L118 has a length of 2, it'll keep the originally-allocated memory.
Preallocation is not strictly necessary but it's a good practice. In this case if there's a chance we'll allocate twice then there's not much of a saving since that's what preallocation is trying to prevent.

Reply:
tried to improve it from your feedback, was it what you had in mind ? 🙏